### DISCUSSION TITLE
Errors handling for Null Class pattern

### DISCUSSION THREAD

**Discussion by thefaded** [TanStack/query#2029] [Q&A]
Hey guys, can you please describe what is the better way to handle errors with using Null Class pattern.
I have this query for instance:
```js
const useAddress = (id) => {
  return useQuery(["address", id], async () => {
    const { data } = await axios.get(`/addresses/${id}`)

    return data
  }, {
    ...DEFAULT_OPTIONS,
    staleTime: DEFAULT_STALE_TIME,
    placeholderData: NullAddress,
  })
}
```
And inside react component using like this:
```js
const { isFetching, isError, data } = useAddress(addressId)
```

But the problem, is what during fetching I'm getting `NullAddress` object – awesome, but once the query fails after 3 attempts, `data` is equal to `undefined`. I have only one solution for this, but it's not looks very good.
```js
const { isFetching, isError, data = NullAddress } = useAddress(addressId)
```

I actually didn't find an option for overriding `undefined` value (the same as `placeholderData` while fetching), am I missing something?

**Comment by TkDodo** [comment]
> but once the query fails after 3 attempts, data is equal to undefined

There is an `error` prop returned from react-query that contains the error. if `isError` is `true`, `error` will be set.
